pkg/costmodel/costmodel.go

```diff
@@ -828,32 +828,56 @@ func findDeletedNodeInfo(cli prometheusClient.Client, missingNodes map[string]*c
 	return nil
 }
 
+// getContainerAllocation takes the max between request and usage. This function
+// returns a slice containing a single element describing the container's
+// allocation. This function assumes that the `req` and `used` slices both
+// contain a single element.
 func getContainerAllocation(req []*util.Vector, used []*util.Vector, allocationType string) []*util.Vector {
```

Comment:
@thomasvn the fix makes sense, and the implementation makes sense — but I'm afraid that keeping this function signature leaves us open to possible misuse in the future by unsuspecting engineers (however unlikely). That is, if we suggest that the function requires a slice of Vectors, but internally we assume that it will have length == 1, and we don't return errors if that condition is not met, then we could see people use this function with a reasonable expectation of the old behavior, and have the function ignore all but the first number. (Also, I want to acknowledge that you've left a helpful comment, but people sometimes don't read comments.)

So, I'd propose we take this a step further in one of two ways:
1. Change the signature here to `getContainerAllocation(req *util.Vector, used *util.Vector, allocationType string) []*util.Vector` so that there can be no confusion. That will require changing the four places this is called, so that we only pass in that first datum, which will itself perhaps require some simple length checking before calling.
2. Change the signature here to `getContainerAllocation(req []*util.Vector, used []*util.Vector, allocationType string) ([]*util.Vector, error)` and return an error if a slice of length > 1 is provided. That will also require changing the four places this is called, to handle the error.

I think that both of these option will help us feel confident in the correctness of this solution, too. (That is, in addition to Ajay's requested unit test.) If we don't see any errors or issues with the new signature, then we can be confident that we won't get bitten by any use cases of which we were not aware.

Reply:
@nikovacevic I agree with changing the function's arguments & signature! Way better than commenting assumptions. Thoughts on the following signature?

```
getContainerAllocation(req *util.Vector, used *util.Vector, allocationType string) *util.Vector {}
```

Reply:
Nevermind the above suggestion I made. `CPUReq`, `CPUUsed`, and `CPUAllocation` are of type `[]*util.Vector`

https://github.com/opencost/opencost/blob/3f62fa0593692c49838f4721d131ffa36eddc91f/pkg/costmodel/costmodel.go#L93-L95

Reply:
Yeah, it's kind of a finicky decision, given the code context and history... I agree that returning `*util.Vector` would be better, but it might be tricky. As long as the function params are singular, I'd be happy! (Or, again, an `error`. Either one.)

Reply:
@nikovacevic Made some changes, when you get a moment could you review again?

Reply:
Will do! Thank you.